src/OpenLoco/src/Ui/Windows/Construction/Common.cpp

```diff
@@ -554,10 +554,85 @@ namespace OpenLoco::Ui::Windows::Construction
     }
 
     // 0x004A6E9B
+    // Update available airports and docks for player company
     void updateAvailableAirportAndDockOptions()
     {
-        // update available airports and docks for player company
-        call(0x004A6E9B);
+        CompanyId currentPlayerCompany = getGameState().playerCompanies[0]; // 004A6E9B
+        CompanyId oldUpdatingCompany = GameCommands::getUpdatingCompanyId();
+        GameCommands::setUpdatingCompanyId(currentPlayerCompany); // 004A6EA0
```

Comment:
This is only used for the duration of this function so we can just scrap this as we can pass the `currentPlayerCompany` instead of the `getUpdatingCompanyId`